test/extended/node_tuning/node_tuning_utils.go

```diff
@@ -0,0 +1,238 @@
+	var (
+		clusterOperators []configv1.ClusterOperator
+		desiredCO        configv1.ClusterOperator
+		isAvailable      bool
+	)
+	clusterOperatorsList, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().List(context.Background(), metav1.ListOptions{})
+	o.Expect(err).ToNot(o.HaveOccurred())
+	clusterOperators = clusterOperatorsList.Items
+	for _, clusterOperator := range clusterOperators {
+		if clusterOperator.Name == coName {
+			desiredCO = clusterOperator
+			e2e.Logf("desiredCO Name is %v", desiredCO.Name)
+			e2e.Logf("desiredCO.status.conditions of %v is %v", desiredCO.Name, desiredCO.Status.Conditions)
+			break
+		}
+	}
+	available := findCondition(desiredCO.Status.Conditions, configv1.OperatorAvailable)
+	degraded := findCondition(desiredCO.Status.Conditions, configv1.OperatorDegraded)
+	progressing := findCondition(desiredCO.Status.Conditions, configv1.OperatorProgressing)
+	e2e.Logf("the status of co %v is: available.Status [%v] degraded.Status [%v] and progressing.Status [%v]", coName, available.Status, degraded.Status, progressing.Status)
+	if available.Status == configv1.ConditionTrue &&
+		degraded.Status == configv1.ConditionFalse &&
+		progressing.Status == configv1.ConditionFalse {
+		e2e.Logf("the status of cluster operator %v restore to updated state", coName)
+		isAvailable = true
+	} else {
+		// print the co status for toubleshooting why the co keep available=false state
+		e2e.Logf("desiredCO status is %v", desiredCO.Status)
+	}
+	return isAvailable, nil
+}
+
+// wait for 10 minutes and check mcp master/worker mcp stay on updated state
+// just in case other test case cause node reboot/scale out or update during execute this test case
+func assertIfSpecifiedMCPSKeepUpdatedStateWithRetry(mcps dynamic.NamespaceableResourceInterface, mcpName string) {
+	errWait := wait.Poll(1*time.Minute, 10*time.Minute, func() (bool, error) {
+		updated, Updating := isPoolUpdated(mcps, mcpName)
+		if updated && !Updating {
+			e2e.Logf("the status of mcp %v is updated state, not updating", mcpName)
+			return true, nil
+		}
+		e2e.Logf("the status of mcp %v is updating or degraded state, will check again")
+		e2e.Logf("the status of mcp is : updated - [%v] updating - [%v]", updated, Updating)
+		return false, nil
+	})
+	AssertWaitPollNoErr(errWait, "the status of mcp keep unexpected state")
+}
+
+// e is return value of Wait.Poll
+// msg is the reason why time out
+// the function assert return value of Wait.Poll, and expect NO error
+// if e is Nil, just pass and nothing happen.
+// if e is not Nil, will not print the default error message "timed out waiting for the condition" because it causes RP AA not to analysis result exactly.
+// if e is "timed out waiting for the condition" or "context deadline exceeded", it is replaced by msg.
+// if e is not "timed out waiting for the condition", it print e and then case fails.
+func AssertWaitPollNoErr(e error, msg string) {
```

Comment:
Does AssertWaitPollNoErr need to be exported? Also, typically, you return the useful data first and then error.  I.e., you probably want to swap the order of what you return.

Reply:
Fixed in new code, thanks!